### DISCUSSION TITLE
Auth implementation details

### DISCUSSION THREAD

**Discussion by yarosdev** [laravel/framework#46147] [Q&A] [answered]
I have checked out a code of auth, regards Session Guard, when using "remember me" it writes to the cookie "recaller " three values: id, token and pwd hash. And my question is why it sends password hash to the browser and even don't use it anywhere ?

**Comment by ImBIOS** [comment]
@yarosdev  Can you provide some code or screenshots?

**Reply by yarosdev** [reply]
![Screenshot_2023-02-19-13-16-39-504_com android chrome](https://user-images.githubusercontent.com/6161950/219944557-66fab3fc-fb22-462c-a771-c7ed2cd5f7ad.jpg)

Here is

**Reply by yarosdev** [reply]
$user->getAuthPassword

**Comment by JurianArie** [comment, marked as answer]
The AuthenticateSession middleware compares the current hash with the hash from the cookie whenever the user was logged in via the recaller.

https://github.com/laravel/framework/blob/48afe79e52924e58de44416840cbbdb31da1df15/src/Illuminate/Session/Middleware/AuthenticateSession.php#L46

**Reply by yarosdev** [reply]
I got it, thank you
